### DISCUSSION TITLE
mfsbsd boot fail: BTX halted

### DISCUSSION THREAD

**Discussion by ramdaron** [ipxe/ipxe#521] [Q&A]
I am trying to load mfsbsd (https://github.com/mmatuska/mfsbsd) version 13.0-STABLE amd64 via iPXE (chainloading using undionly.kpxe, ver. 2265a65) on Intel D525MWV motherboard and get "BTX halted" error. All previous versions of FreeBSD are loaded without problems.
Settings in BIOS:
```
UEFI Boot: Disable
Boot to Network: Enable PXE
```
But when I switch "Enable PXE" to "Enable gPXE" - the OS boots normally.

If I understood the contents of these two links correctly (https://docs.freebsd.org/en/books/arch-handbook/boot/#btx-server and https://people.freebsd.org/~jhb/docs/loader.txt section 24.3.1.2), the problem is in the changes in the FreeBSD loader code.
My question is: what changes for the FreeBSD loader depending on the selected option in the BIOS (pxe/gpxe) when booting using ipxe?

**Comment by NiKiZe** [comment]
You should probably ask that to the mfsbsd/bsd people.
I wonder if there is any difference that iPXE can detect, try loading the different binaries into iPXE and run imgstat and post output?

**Comment by ramdaron** [comment]
I'm not sure, but I think that ipxe creates its environment via the udionly.kpxe boot and FreeBSD doesn't know anything about PXE/gPXE in BIOS.

My ipxe file:
```
#!ipxe
imgfree
kernel http://10.0.1.1/memdisk iso raw
initrd http://10.0.1.1/iso/BSD/mfsbsd-13.0-RELEASE-amd64.iso
shell
```

imgstat output:
```
memdisk : 26140 bytes [bzImage] [SELECTED] "iso raw"
mfsbsd-13.0-RELEASE-amd64.iso : 136323072 bytes
```
And there is no difference between imgstat output for download via PXE and gPXE in BIOS.
syslinux version 6.03

**Reply by NiKiZe** [reply]
You are in this case emulating a cd in memory, iPXE just transferres execution to memdisk which listens calls to disk thru the INT13 BIOS calls, but as soon as any other kernel is looded, those INT13 calls are out the window.

When iPXE loads any non NBP image, it also unloads the undi stack.

In the end, you are not trying to make the boot PXE aware, you are just booting a emulated CD.
